app/src/main/kotlin/com/simplemobiletools/gallery/helpers/MediaSideScroll.kt

```diff
@@ -0,0 +1,150 @@
+package com.simplemobiletools.gallery.helpers
+
+import android.app.Activity
+import android.content.Context
+import android.media.AudioManager
+import android.os.Handler
+import android.provider.Settings
+import android.view.MotionEvent
+import android.widget.TextView
+import com.simplemobiletools.gallery.R
+import com.simplemobiletools.gallery.activities.ViewPagerActivity
+import com.simplemobiletools.gallery.extensions.audioManager
+
+class MediaSideScroll(val context: Context, val activity: Activity, val slideInfoView: TextView, val onSpecialEvent: () -> Unit) {
+    private val CLICK_MAX_DURATION = 150
+    private val SLIDE_INFO_FADE_DELAY = 1000L
+    private var mTouchDownX = 0f
+    private var mTouchDownY = 0f
+    private var mTouchDownTime = 0L
+    private var mTouchDownVolume = 0
+    private var mTouchDownBrightness = -1
+    private var mTempBrightness = 0
+    private var mLastTouchY = 0f
+
+    private var mSlideInfoText = ""
+    private var mSlideInfoFadeHandler = Handler()
+
+    fun handleVolumeTouched(event: MotionEvent) {
+        when (event.action) {
+            MotionEvent.ACTION_DOWN -> {
+                mTouchDownX = event.x
+                mTouchDownY = event.y
+                mLastTouchY = event.y
+                mTouchDownTime = System.currentTimeMillis()
+                mTouchDownVolume = getCurrentVolume()
+                mSlideInfoText = "${context.getString(R.string.volume)}:\n"
+            }
+            MotionEvent.ACTION_MOVE -> {
+                val diffX = mTouchDownX - event.x
+                val diffY = mTouchDownY - event.y
+
+                if (Math.abs(diffY) > 20 && Math.abs(diffY) > Math.abs(diffX)) {
+                    var percent = ((diffY / ViewPagerActivity.screenHeight) * 100).toInt() * 3
+                    percent = Math.min(100, Math.max(-100, percent))
+
+                    if ((percent == 100 && event.y > mLastTouchY) || (percent == -100 && event.y < mLastTouchY)) {
+                        mTouchDownY = event.y
+                        mTouchDownVolume = getCurrentVolume()
+                    }
+
+                    volumePercentChanged(percent)
+                }
+                mLastTouchY = event.y
+            }
+            MotionEvent.ACTION_UP, MotionEvent.ACTION_CANCEL -> {
+                val diffX = Math.abs(event.x - mTouchDownX)
+                val diffY = Math.abs(event.y - mTouchDownY)
+                if (System.currentTimeMillis() - mTouchDownTime < CLICK_MAX_DURATION && diffX < 20 && diffY < 20) {
+                    onSpecialEvent()
```

Comment:
Not sure how to call the callback, what does the condition check?

Reply:
Also, should I split the photo/video functionality into two separate classes? This callback is unused in PhotoFragment. But I do like the idea of having all the volume & brightness functionality in one class, just need to move this callback out somehow. If only I could know what the condition does.

Reply:
it checks for clicks. You still have to be able to click through the slider

Reply:
for the callback, add some addCallback(callback) function or so and store it in a variable